core/NameRef.h

```diff
@@ -78,13 +78,22 @@ class NameRef final : private DebugOnlyCheck<NameRefDebugCheck> {
 
     const NameData data(const GlobalState &gs) const;
 
+    // Returns the `-1` NameRef, used to indicate that a NameRef hasn't been written to (initialized)
+    static NameRef uninitialized() {
+        return NameRef(WellKnown{}, -1);
+    }
+
     // Returns the `0` NameRef, used to indicate non-existence of a name
     static NameRef noName() {
         return NameRef(WellKnown{}, 0);
     }
 
+    inline bool isInitialized() const {
+        return _id != -1;
+    }
+
     inline bool exists() const {
-        return _id != 0;
+        return _id > 0;
```

Comment:
Another thing we could do here is ENFORCE if you call `.exists()` on a Name before it's initialized, but I think this is more convenient (for example, `SymbolRef sym; sym.exists();` returns `false` instead of raising).